applications/hamster.md

```diff
@@ -0,0 +1,278 @@
+| 2. | Hamster Node: pallet_gateway | We will create a Substrate module that will have **Register gateway**,**Receive rewards** etc function. |
+| 3. | Hamster Gateway: web app | We will create a web app integrated in the Hamster Gateway, to let users easily interact with our Hamster Gateway function module.<br/>Tech stack: vue.js+polkadot.js+Node.js |
+| 4. | Hamster chain | Add a gateway module in the Hamster chain, to enable users to share gateway resources with public IP , receive rewards, etc. |
+
+### ~~Milestone 3 Example — Implement Hamster Anti-malicious attacks Modules~~
+
+- ~~**Estimated duration:** 1 month~~
+- ~~**FTE:**  4~~
+- ~~**Costs:** $12,800~~
+
+| ~~Number~~ | ~~Deliverable~~ | ~~Specification~~ |
+|-------:| ----------- | ------------- |
+| ~~0a.~~ | ~~License~~ | ~~Apache 2.0~~ |
+| ~~0b.~~ | ~~Documentation~~ | ~~We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can use Hamster to share and use computing resources~~ |
+| ~~0c.~~ | ~~Testing Guide~~ | ~~Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.~~ |
+| ~~0d.~~ | ~~Article~~ | ~~We will write an article or tutorial that explains the work done as part of the grant.~~ |
+| ~~1.~~ | ~~Evil punishment~~ | ~~Design and implementation of punishment module for evil doing~~ |
+| ~~2.~~ | ~~Hamster Front end~~ | ~~Front end function addition and optimization to let users easily interact with our Hamster function module.~~ |
+| ~~3.~~ | ~~Benchmark~~ | ~~Perform unit tests on the individual algorithms to ensure system safety.~~ |
+
+## Future Plans
+
+Our current goal is to provide distributed shared computing power. After this milestone is completed, we will be able to build our own ecosystem based on resources. That is, when everyone is willing to add their free resources to the hamster network, it can be assumed that there are near-infinite computing resources in the network, and more quality services can be built and constructed in the hamster network. For example, service providers no longer need to rely on the support of a single cloud vendor, but only need to use the computing resources in the hamster network, and the system automatically schedules the resource arithmetic needed for the services to build their own services in the form of edge computing. In addition to this, we can also provide some toolkits so that these computing resources can participate in other services with one click, such as becoming a chain node of Thegraph, so that they can participate in other networks, and later on, we can also use incentives to encourage people to develop toolkits for different ecologies. When more and more people and more service providers participate in the hamster network, maybe it can become a truly decentralized meta-universe service cornerstone.
```

Comment:
> we can also use incentives to encourage people to develop toolkits for different ecologies.

What do you mean by "different ecologies"? Or did you mean "different technologies" instead?

> maybe it can become a truly decentralized meta-universe service cornerstone.

I'm not sure what that means.

Reply:
Sorry, we may not have made it clear that "different ecologies" here means that we can provide computing power to other projects in the blockchain ecosystem that need computing resources. And we hope to support the development of plugins that allow our computing resources to provide services directly, so that real users can quickly support their favorite blockchain projects.

Regarding the cornerstone of the meta-verse service, in Chinese we call it the infrastructure of the meta-verse. I think the meta-verse is virtualized in our world by consuming a lot of computational resources to accomplish this. Computational resources are the basis for carrying the meta-verse computation, and I think maybe the computational nodes of the meta-verse are mega arithmetic centers and supercomputer nodes etc. for providing. Perhaps I have the luxury of hoping that our computing services will reach a certain scale and we can provide our computing power in the meta-verse as well.

Reply:
Yes, you're right, that's what we're trying to say, thank you very much